### DISCUSSION TITLE
OmniSharp Not Working

### DISCUSSION THREAD

**Discussion by BButner** [helix-editor/helix#5849] [Q&A] [answered]
I'm attempting to start using Helix for the first time, and attempting to use it on my dotnet codebase.

I've gotten Helix installed via choco, then downloaded OmniSharp (NOT the http version, the net6.0 version specifically), and added that to my path.

When I open up Windows Terminal and I run `hx --health c-sharp` I get:

![image](https://user-images.githubusercontent.com/39167186/217027979-3c50dbe2-acb4-4bc9-ae03-882d4506e1fa.png)

Notice that Highlight and Textobject are both not checked.

When I open a .cs file, I get this in the log file:

![image](https://user-images.githubusercontent.com/39167186/217028327-c91cddc3-8320-4604-95eb-23f999303bf4.png)

However, if I run Windows Terminal as administrator, I get this running the `hx --health c-sharp` command:

![image](https://user-images.githubusercontent.com/39167186/217028484-097e4956-e974-45e1-9f84-efa75ce2d0ee.png)

But I get the same error messages in the log...

If I try to use the *non* net6.0 version, I get absolutely nothing in the log when opening a .cs file.

**Comment by pascalkuthe** [comment, marked as answer]
I would suggest that you use either the precompiled binaries from the [release page](https://github.com/helix-editor/helix/releases/tag/22.12) as the chocolatey and scoop package is quite outdated at the moment (it's still using `22.08` instead of the newer `22.12` release which came out 2 months ago). Helix is developing very quickly and using older versions you will miss out on many improvements, features and bugfixes. You can follow the instructions for linking the runtime directory the documentation for building from source (just that you can use the downloaded binaries instead of actually building from source): https://docs.helix-editor.com/install.html

I am not quite sure what your issue is as you are just saying that the LS is not working but not actually specifying what exactly is not working. You are mostly talking about highlight queries and textobjects afterwards so I assume that those are not working.

Language servers only provide autocompletion, goto defintion etc. so the language server might work just fine for you. However it does seem that these is a problem with your tree sitter grammars (these provider syntax highlights and textobject queries). These are part of the runtime directory described in the README and should work out of the box if helix was installed correctly. Perhaphs just installing from scoop will fix the problem.

**Reply by BButner** [reply]
Cloning the repo and compiling that resolved a lot of the LSP issues that I was having, thanks for the suggestion! `hx --version` is now reporting `helix 22.12 (9c98043c)`.

OmniSharp was still not working out of the box, until I changed the configuration to the below:

```toml
[[language]]
name = "c-sharp"
language-server = { command = "dotnet", args = [ "C:/omnisharp-win-x64-net6.0/OmniSharp.dll", "--languageserver" ] }
```

That being said, I'm running into an issue with the LSP in a larger project. When I open Helix in the root directory then open one of the files, it starts loading all the sub-projects in the project. There are 100+ projects. Running Helix with `-v`, I can see OmniSharp loading all of the projects just fine. Looking further into the log, I see Helix reporting:

 `2023-02-06T13:48:29.529 helix_lsp [ERROR] failed to initialize language server: request timed out`

However, the OmniSharp LSP is still actively loading and successfully loading projects. If I go to my `languages.toml` and set a massive timeout, the LSP continues to load and eventually functions just fine. It seems as if Helix is timing out the LSP, even though its actively loading projects?

**Reply by the-mikedavis** [reply]
Yeah, the timeout applies to all LSP requests: we time out and consider the request failed if we don't receive a response in a set amount of time. (LSP initialize/initialized is a request/response pair.) That is intended for cases where the language server never responds. For cases where the language server can take a really long time, increasing the timeout in the `languages.toml` is the right choice :+1:
